homeassistant/components/switch/homematicip_cloud.py

```diff
@@ -0,0 +1,81 @@
+"""
+Support for HomematicIP switch.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/switch.homematicip_cloud/
+"""
+
+import logging
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.components.homematicip_cloud import (
+    HomematicipGenericDevice, DOMAIN as HOMEMATICIP_CLOUD_DOMAIN,
+    ATTR_HOME_ID)
+
+DEPENDENCIES = ['homematicip_cloud']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_POWER_CONSUMPTION = 'power_consumption'
+ATTR_ENERGIE_COUNTER = 'energie_counter'
+ATTR_PROFILE_MODE = 'profile_mode'
+
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
+    """Set up the HomematicIP switch devices."""
+    from homematicip.device import (
+        PlugableSwitch, PlugableSwitchMeasuring,
+        BrandSwitchMeasuring)
+
+    if discovery_info is None:
+        return
+    home = hass.data[HOMEMATICIP_CLOUD_DOMAIN][discovery_info[ATTR_HOME_ID]]
+    devices = []
+    for device in home.devices:
+        if isinstance(device, BrandSwitchMeasuring):
+            pass
```

Comment:
Why are you implementing this as a light and not as a switch ?

Reply:
@sander76 Yes, this was the idea... do you agree?

Reply:
@sander76 So far I would have the pluggable switches (+ measuring) into the "switch" components and the devices like brand wall light switches (+ measuring & dimming devices) into the "light" components.

Reply:
@mxworm I guess in the end it doesn't really matter to hass anyway..